cmd/mayactl/app/command/volume_info.go

```diff
@@ -166,63 +203,66 @@ Status  :   {{.Status}}
+		replicaCount, _ = strconv.Atoi(v.GetReplicaCount())
+		// This case will occur only if user has manually specified zero replica.
+		if replicaCount == 0 || len(v.GetReplicaStatus()) == 0 {
+			fmt.Println("None of the replicas are running, please check the volume pod's status by running [kubectl describe pod -l=openebs/replica --all-namespaces] or try again later.")
+			return nil
 		}
-	}
+		// Splitting strings with delimiter ','
+		replicaStatusStrings := strings.Split(v.GetReplicaStatus(), ",")
+		addressIPStrings := strings.Split(v.GetReplicaIP(), ",")
 
-	// We get the info of the running replicas from the collection.data.
-	// We are appending modes if available in collection.data to replicaIPStatus
+		// making a map of replica ip and their respective status,index and mode
+		replicaIPStatus := make(map[string]*Value)
 
-	replicaInfo := make(map[int]*ReplicaInfo)
-	for key := range collection.Data {
-		address = append(address, strings.TrimSuffix(strings.TrimPrefix(collection.Data[key].Address, "tcp://"), v1.ReplicaPort))
-		mode = append(mode, collection.Data[key].Mode)
-		if _, ok := replicaIPStatus[address[key]]; ok {
-			replicaIPStatus[address[key]].mode = mode[key]
+		// Creating a map of address and mode. The IP is chosed as key so that the status of that corresponding replica can be merged in linear time complexity
+		for index, IP := range addressIPStrings {
+			if IP != "nil" {
+				replicaIPStatus[IP] = &Value{index: index, status: replicaStatusStrings[index], mode: "NA"}
+			} else {
+				// appending address with index to avoid same key conflict as the IP is returned as `nil` in case of error
+				replicaIPStatus[IP+string(index)] = &Value{index: index, status: replicaStatusStrings[index], mode: "NA"}
+			}
 		}
-	}
 
-	for k, v := range replicaIPStatus {
-		// checking if the first three letters is nil or not if it is nil then the ip is not avaiable
-		if k[0:3] != "nil" {
-			replicaInfo[v.index] = &ReplicaInfo{k, v.mode, v.status, "NA", "NA"}
-		} else {
-			replicaInfo[v.index] = &ReplicaInfo{"NA", v.mode, v.status, "NA", "NA"}
+		// We get the info of the running replicas from the collection.data.
+		// We are appending modes if available in collection.data to replicaIPStatus
+		replicaInfo := make(map[int]*ReplicaInfo)
+
+		for key := range collection.Data {
+			address = append(address, strings.TrimSuffix(strings.TrimPrefix(collection.Data[key].Address, "tcp://"), v1.ReplicaPort))
+			mode = append(mode, collection.Data[key].Mode)
+			if _, ok := replicaIPStatus[address[key]]; ok {
+				replicaIPStatus[address[key]].mode = mode[key]
+			}
 		}
-	}
 
-	err = updateReplicasInfo(replicaInfo)
-	if err != nil {
-		fmt.Println("Error in getting specific information from K8s. Please try again.")
-	}
+		for IP, replicaStatus := range replicaIPStatus {
+			// checking if the first three letters is nil or not if it is nil then the ip is not avaiable
+			if IP[0:3] != "nil" {
```

Comment:
Not the best approach. use strings contains.

Reply:
done